pkg/build/builder/sti.go

```diff
@@ -226,12 +228,11 @@ func (s *S2IBuilder) Build() error {
 
 	// If DockerCfgPath is provided in buildapiv1.Config, then attempt to read the
 	// dockercfg file and get the authentication for pulling the images.
-	t, _ := dockercfg.NewHelper().GetDockerAuth(config.BuilderImage, dockercfg.PullAuthType)
-	config.PullAuthentication = s2iapi.AuthConfig{Username: t.Username, Password: t.Password, Email: t.Email, ServerAddress: t.ServerAddress}
 
 	if s.build.Spec.Strategy.SourceStrategy.ForcePull || !isImagePresent(s.dockerClient, config.BuilderImage) {
 		startTime := metav1.Now()
-		err = s.pullImage(config.BuilderImage, t)
+		auth, _ := dockercfg.NewHelper().GetAllDockerAuths(dockercfg.PullAuthType)
```

Comment:
returning `error` instead of `bool` is instructive here. In this case we probably would not want to return error if the auth config could not be read. We'd fall back to pulling an image anonymously.

Reply:
in the current implementation we only return false when an error occurred so they are equivalent in that sense ... we no longer care which auths are present, and are in fact still trying to pull anonymously

in the other file, the use of authPresent ultimately was only used for triggering some debug

Reply:
again no-op with new changes